code/modules/clothing/suits/_suits.dm

```diff
@@ -4,18 +4,172 @@
 	lefthand_file = 'icons/mob/inhands/clothing/suits_lefthand.dmi'
 	righthand_file = 'icons/mob/inhands/clothing/suits_righthand.dmi'
 	var/fire_resist = T0C+100
-	allowed = list(
-		/obj/item/tank/internals/emergency_oxygen,
-		/obj/item/tank/internals/plasmaman,
-		/obj/item/tank/jetpack/oxygen/captain,
-		/obj/item/storage/belt/holster,
-		)
 	armor_type = /datum/armor/none
 	drop_sound = 'sound/items/handling/cloth_drop.ogg'
 	pickup_sound = 'sound/items/handling/cloth_pickup.ogg'
 	slot_flags = ITEM_SLOT_OCLOTHING
 	var/blood_overlay_type = "suit"
 	limb_integrity = 0 // disabled for most exo-suits
+	allowed = list(
```

Comment:
Also - no matter what happens, you should just outright delete the `allowed` var and have a global typecache. There's no need to have this massive list on every single suit item

Reply:
Or I guess if it's still used in some places, then put this mega-list in a global typecache and check `allowed || GLOB.suit_allowables` wherever relevant

Reply:
maybe... im pretty new to byond stuff but i could look into it

Reply:
Ask if you need help but as it stands putting a 150 length element list on every suit item is a bit of a waste of memory

Reply:
yeah, id appreciate help on that x3
and im open to nerfing the gun stuff, it's not really the main point of the PR